### PR TITLE
Refactor settings

### PR DESCRIPTION
**Pull request by mstv**
## Proposed changes

Extract refactorings from draft PRs 
- Rename to `DistributedSettings`
- Remove unused property `ConfigFile.Local`
- Rename argument `allowCache`
- Name arguments
- Use explicit types
- Declare `SettingsSet`s as `readonly record struct`
- `SettingsPageHeader`: Rename arrow labels; remove `TabIndex`
- Add / synchronize `DebuggerDisplay` attributes

## Screenshots 

no change

## Test methodology 

- existing tests

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/CommandsDialogs/SettingsDialog/RepoDistSettingsSet.cs [hunk 1]
```diff
@@ -2,23 +2,9 @@
...
-            RepoDistSettings pulledSettings,
-            RepoDistSettings globalSettings)
-        {
-            EffectiveSettings = effectiveSettings;
-            LocalSettings = localSettings;
-            RepoDistSettings = pulledSettings;
-            GlobalSettings = globalSettings;
-        }
-    }
+    public readonly record struct RepoDistSettingsSet(
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/SettingsDialog/RepoDistSettingsSet.cs:5]: Rename this struct too?
```suggestion
    public readonly record struct DistributedSettingsSet(
```

**Review by gerhardol — APPROVED** [review]

File: GitCommands/Settings/ConfigFileSettings.cs [hunk 2]
```diff
@@ -5,51 +5,53 @@
 
 namespace GitCommands.Settings
 {
+    [DebuggerDisplay("{" + nameof(SettingLevel) + "} {" + nameof(SettingsCache) + "} << {" + nameof(LowerPriority) + "}")]
```
Comment (RussKie) [inline GitCommands/Settings/ConfigFileSettings.cs:8]: Do you think you could add a screenshot of how it looks?

File: GitCommands/Settings/ConfigFileSettingsCache.cs [hunk 3]
```diff
@@ -1,35 +1,37 @@
-﻿using GitCommands.Config;
+﻿using System.Diagnostics;
+using GitCommands.Config;
 using GitUIPluginInterfaces;
 
 namespace GitCommands.Settings
 {
+    [DebuggerDisplay("{_byNameMap.Count} {" + nameof(SettingsFilePath) + ",nq}")]
```
Comment (RussKie) [inline GitCommands/Settings/ConfigFileSettingsCache.cs:7]: Ditto here

File: UnitTests/GitCommands.Tests/Config/ConfigFileTest.cs [hunk 4]
```diff
@@ -74,7 +74,7 @@ public void TestWithInvalidFileName()
                 File.WriteAllText(GetConfigFileName(), GetDefaultConfigFileContent(), GitModule.SystemEncoding);
             }
 
-            ConfigFile configFile = new(GetConfigFileName() + "\\", false);
+            ConfigFile configFile = new(GetConfigFileName() + "\\");
```
Comment (RussKie) [inline UnitTests/GitCommands.Tests/Config/ConfigFileTest.cs:77]: Can/should we use the helper method "EnsureTrailingSlash" instead?

File: GitUI/BuildServerIntegration/BuildServerWatcher.cs [hunk 5]
```diff
@@ -155,7 +155,7 @@ public void CancelBuildStatusFetchOperation()
                             byte[] unprotectedData = ProtectedData.Unprotect(protectedData, null,
                                 DataProtectionScope.CurrentUser);
                             using MemoryStream memoryStream = new(unprotectedData);
-                            ConfigFile credentialsConfig = new("", false);
+                            ConfigFile credentialsConfig = new("");
```
Comment (RussKie) [inline GitUI/BuildServerIntegration/BuildServerWatcher.cs:158]: [nit] Not terribly important, but should this be?
```suggestion
                            ConfigFile credentialsConfig = new(fileName: "");
```

Here and in few other places

**Review by RussKie — APPROVED** [review]

File: GitCommands/Settings/ConfigFileSettings.cs (hunk 2 above)
Comment (mstv) [reply GitCommands/Settings/ConfigFileSettings.cs:8]: image

File: GitCommands/Settings/ConfigFileSettingsCache.cs (hunk 3 above)
Comment (mstv) [reply GitCommands/Settings/ConfigFileSettingsCache.cs:7]: see screenshot above

File: UnitTests/GitCommands.Tests/Config/ConfigFileTest.cs (hunk 4 above)
Comment (mstv) [reply UnitTests/GitCommands.Tests/Config/ConfigFileTest.cs:77]: > Can/should we use the helper method "EnsureTrailingSlash" instead?

This under full control of this test:
```
        private static string GetConfigFileName()
        {
            return Path.Combine(GetTempFolder(), "testconfigfile");
        }
```

File: GitUI/BuildServerIntegration/BuildServerWatcher.cs (hunk 5 above)
Comment (mstv) [reply GitUI/BuildServerIntegration/BuildServerWatcher.cs]: Added for the three occurrences (excluding tests). It is Config**File**, so indeed "not terribly important".

File: UnitTests/GitCommands.Tests/Config/ConfigFileTest.cs (hunk 4 above)
Comment (RussKie) [reply UnitTests/GitCommands.Tests/Config/ConfigFileTest.cs:77]: Ah, didn't pay attention that it was in the test

File: GitCommands/Settings/ConfigFileSettings.cs (hunk 2 above)
Comment (RussKie) [reply GitCommands/Settings/ConfigFileSettings.cs:8]: What would this look like?
```suggestion
    [DebuggerDisplay("{" + nameof(SettingLevel) + "} ({" + nameof(SettingsCache) + "}) <- {" + nameof(LowerPriority) + "}")]
```

File: GitCommands/Settings/ConfigFileSettingsCache.cs (hunk 3 above)
Comment (RussKie) [inline GitCommands/Settings/ConfigFileSettingsCache.cs:7]:
```suggestion
    [DebuggerDisplay("Values: {_byNameMap.Count}, File: {" + nameof(SettingsFilePath) + ",nq}")]
```

Comment (mstv) [reply GitCommands/Settings/ConfigFileSettingsCache.cs:7]: - "<<" looks better than "<-" and matches the Settings dialog.
- I can add parens for `Count`.
- The resulting strings are too long already. The labels are pretty obvious if one is debugging this code.

File: GitCommands/Settings/ConfigFileSettings.cs (hunk 2 above)
Comment (mstv) [reply GitCommands/Settings/ConfigFileSettings.cs:8]: > What would this look like?

Only if you really insist:

image

or:

image

Comment (mstv) [reply GitCommands/Settings/ConfigFileSettings.cs:8]: Igor, do you accept
image
?

**Pull request merged**